### PR TITLE
Allow Http/Https dyanmic switching of loading gmap

### PR DESCRIPTION
**Pull request by mlzummo**
Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

File: src/map/GMap.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 "use strict";
 (function (root, factory) {
     if (typeof define === "function" && define.amd) {
-        define(["d3", "../common/HTMLWidget", "../layout/AbsoluteSurface", "async!http://maps.google.com/maps/api/js?sensor=false", "css!./GMap"], factory);
```
Comment (GordonSmith) [inline src/map/GMap.js]: Did search for anywhere else we may be making this mistake?

File: src/map/GMap.js [hunk 2]
```diff
@@ -29,7 +29,7 @@
         this._prevMin = { x: 0, y: 0 };
         this._prevMax = { x: 0, y: 0 };
     }
-    Overlay.prototype = new google.maps.OverlayView();
+    Overlay.prototype = new google.maps.OverlayView() || null;
```
Comment (GordonSmith) [inline src/map/GMap.js]: ```|| null``` makes no sense...

Comment (mlzummo) [inline src/map/GMap.js]: I was suppose to remove that mod ... I'm trying to figure out how to fail
gracefully

**Comment by mlzummo** [comment]
seems like the unit tests didnt like this too much ...

File: src/map/GMap.js (hunk 1 above)
Comment (GordonSmith) [inline src/map/GMap.js]: @mlzummo does "async!//maps.google.com/maps/api/js?sensor=false" work?

Comment (mlzummo) [inline src/map/GMap.js]: it will fail in unit tests and when running locally if you do it with \\

Comment (GordonSmith) [inline src/map/GMap.js]: Ok defaulting to ```file:```  I suspect?

**Comment by mlzummo** [comment]
@GordonSmith this is fixed ... can you pull?

**Pull request merged**